x-pack/plugin/data-streams/src/main/java/org/elasticsearch/xpack/datastreams/action/DeleteDataStreamTransportAction.java

```diff
@@ -86,7 +86,6 @@ protected void masterOperation(
         for (String name : names) {
             systemIndices.validateDataStreamAccess(name, threadPool.getThreadContext());
         }
-        request.indices(names.toArray(Strings.EMPTY_ARRAY));
 
```

Comment:
I'm not sure that it's useful to duplicate this logic in `masterOperation` and below in `removeDataStream` but the overwriting of the request's indices is the most narrow root cause of this bug.

Reply:
I think it's fine to double-check the validation so that we avoid the cluster state task if possible, so this seems reasonable to me.